src/utils/readOnlyUtils.ts

```diff
@@ -0,0 +1,59 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+/*
+    This functionality is more or less "cherry-picked" from package "vscode-azureextensionui".
+    Currently we can not use it from there due to the fact that we depend on an old version.
+    Once we move to a more recent version we should eventually remove this file and rewire the
+    call sites to the "genuine" implementation.
+*/
+
+import { CancellationToken, Event, EventEmitter, TextDocumentContentProvider, Uri, window, workspace } from "vscode";
+import { createHash} from "crypto";
+import { isNullOrUndefined } from "util";
+import { ext } from "../extensionVariables";
+
+let contentProvider: ReadOnlyContentProvider | undefined;
+const providerScheme = 'azurelogicappsReadonly';
+
+export async function openReadOnlyJson(name: string, content: string) {
+    return openReadOnlyContent(name, content, '.json');
+}
+
+export async function openReadOnlyContent(name: string, content: string, fileExt: string) {
+    if (!contentProvider) {
+        contentProvider = new ReadOnlyContentProvider();
+        ext.context.subscriptions.push(
+            workspace.registerTextDocumentContentProvider(providerScheme, contentProvider));
+    }
+    await contentProvider.openReadOnlyContent(name, content, fileExt);
+}
+
+class ReadOnlyContentProvider implements TextDocumentContentProvider {
+    private _onDidChangeEmitter = new EventEmitter<Uri>();
+    private _contentMap = new Map<string, string>();
+
+    public get onDidChange(): Event<Uri> {
+        return this._onDidChangeEmitter.event;
+    }
+
+    public async openReadOnlyContent(name: string, content: string, fileExt: string) {
+        const nameHash  = createHash('sha256').update(name).digest('hex');
+        const uri = Uri.parse(`${providerScheme}:///${nameHash}/${name}${fileExt}`);
```

Comment:
Is it intentional to have three slashes here instead of two?

Reply:
This was brought in from the original [implementation](https://github.com/microsoft/vscode-azuretools/blob/e261027bb70893a08210b81585419a84a5c0d7d4/ui/src/openReadOnlyContent.ts#L51)
So I don't know whether they did it intentionally but I think so